drivers/spi/spi_rpi_pico_pio.c

```diff
@@ -23,20 +23,44 @@ LOG_MODULE_REGISTER(spi_pico_pio);
 #include <hardware/pio.h>
 #include "hardware/clocks.h"
 
+#if defined(CONFIG_SPI_RPI_PICO_PIO_DMA)
+
+#include <zephyr/drivers/dma.h>
+#include <zephyr/spinlock.h>
+#if defined(CONFIG_SOC_SERIES_RP2040)
+#include <zephyr/dt-bindings/dma/rpi-pico-dma-rp2040.h>
+#elif defined(CONFIG_SOC_SERIES_RP2350)
+#include <zephyr/dt-bindings/dma/rpi-pico-dma-rp2350.h>
+#endif
+
+#include "hardware/dma.h"
+
+#endif
+
 #define SPI_RPI_PICO_PIO_HALF_DUPLEX_ENABLED DT_ANY_INST_HAS_PROP_STATUS_OKAY(sio_gpios)
 
 #define PIO_CYCLES     (4)
 #define PIO_FIFO_DEPTH (4)
 
+struct spi_pico_pio_dma_config {
+	const struct device *dev;
+	uint32_t tx_channel;
+	uint32_t rx_channel;
+};
+
 struct spi_pico_pio_config {
 	const struct device *piodev;
 	const struct pinctrl_dev_config *pin_cfg;
+	const bool dma_enabled;
```

Comment:
Do you really need this? is dma_config.dev is non NULL it would be the same no? 
(this bool is misaligning your struct thus why there is perhaps a way to not introduce it)

Reply:
Yes, I agree with you. I was just mimicking what was done in the pl022 driver, and I think your comment will apply to the pl022 driver equally well.